### DISCUSSION TITLE
Please help me with this bug, I don't have any clue!!

### DISCUSSION THREAD

**Discussion by willyhuang18** [vercel/vercel#8648] [Help]
I think my files are ok, I am using Next.js and Sanity as database.
So I tried to run "vercel deploy" on terminal, and I got this error, the file it mentions, I checked it, and it actually exists. 
"Vercel build" is working but not the deploy. Please Help !! Thank you!!!

![image](https://user-images.githubusercontent.com/87446864/192678910-33a8f4cd-3d74-4440-b0db-fc4276e33a3c.png)

**Comment by amyegan** [comment]
Hi @willyhuang18! If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I would be happy to help you find the source of the problem

**Reply by willyhuang18** [reply]
I am not sure what you need, but here is the repo https://github.com/willyhuang18/Front-End-PlayGround/tree/master/will-portfolio, the application is run functional in localhost300

**Reply by amyegan** [reply]
Thanks for sharing the code! Someone else recently had the [same problem](https://github.com/vercel/vercel/discussions/8641) and solved it by changing `getStaticProps` to `getServerSideProps` in [`/pages/index.tsx`](https://github.com/willyhuang18/Front-End-PlayGround/blob/ef3191152a4e8cd785d83f9ea0448fcbbc61e1bc/will-portfolio/pages/index.tsx#L78).

Is this based on a tutorial or starter template that you can share? I would like to update the original source to prevent other people from running into the same error again 😃

**Reply by willyhuang18** [reply]
It worked somehow, Thank you. but the application I deployed,  has code 500, Internal Server Error. It said that missing projectid. but in my .env file, it has the id. I am able to print out the ProjectID. I think we both followed the tutorial on Youtube, the YouTuber was using getStaticProps. but when he deployed, he didn't have any errors on his side. 
![image](https://user-images.githubusercontent.com/87446864/192894033-3bbbebc1-af54-4a50-9289-0d188e6c1303.png)
